src/app/qgsmaptooladdabstract.cpp

```diff
@@ -0,0 +1,119 @@
+/***************************************************************************
+    qgsmaptooladdabstract.cpp  -  abstract class for map tools of the 'add' kind
+    ---------------------
+    begin                : July 2017
+    copyright            : (C) 2017
+    email                : lbartoletti at tuxfamily dot org
```

Comment:
Credit should go to you ;)
I've been wanting to do this refactoring for a long time. Thank you.